engine-standalone-tracing/src/sputnik.rs

```diff
@@ -24,6 +30,165 @@ where
+                            if !return_value.is_empty() {
+                                self.output = return_value.to_vec();
+                            }
+                            if !reason.is_succeed() {
+                                self.failed = true;
+                            }
+                        }
+                    }
+                    Err(Capture::Trap(opcode)) => {
+                        // "Trap" here means that there is some opcode which has special
+                        // handling logic outside the core `step` function. This means the
+                        // `StepResult` does not necessarily indicate the current log
+                        // is finished yet. In particular, `SLoad` and `SStore` events come
+                        // _after_ the `StepResult`, but still correspond to the current step.
+                        if opcode == &Opcode::SLOAD || opcode == &Opcode::SSTORE {
+                            // will push the log after processing `SLOAD` / `SSTORE` events
+                        } else {
+                            self.logs.push(self.current.clone());
+                        }
+                    }
+                }
+            }
+
+            Event::SLoad {
+                address: _,
+                index,
+                value,
+            } => {
+                self.current
+                    .storage
+                    .insert(LogStorageKey(index.0), LogStorageValue(value.0));
+                self.logs.push(self.current.clone());
+            }
+
+            Event::SStore {
+                address: _,
+                index,
+                value,
+            } => {
+                self.current
+                    .storage
+                    .insert(LogStorageKey(index.0), LogStorageValue(value.0));
+                self.logs.push(self.current.clone());
+            }
+        }
+    }
+}
+
+impl evm::tracing::EventListener for TransactionTraceBuilder {
+    fn event(&mut self, event: evm::tracing::Event) {
+        use evm::tracing::Event;
+        match event {
+            Event::Call { .. } => {
+                self.current.depth.increment();
+            }
+            Event::Create { .. } => {
+                self.current.depth.increment();
+            }
+            Event::Suicide { .. } => (), // TODO: ???
+            Event::Exit {
+                reason: _,
+                return_value,
+            } => {
+                if !self.current.depth.is_zero() {
+                    // If the depth is not zero then an error must have occurred to
+                    // exit early.
+                    self.failed = true;
+                    self.output = return_value.to_vec();
+                }
+            }
+            Event::TransactCall { .. } => (), // no useful information
+            Event::TransactCreate { .. } => (), // no useful information
+            Event::TransactCreate2 { .. } => (), // no useful information
```

Comment:
This could be represented simple as `_ => (),` to shorten the code here.

Reply:
I like explicitly listing all the cases because then the compiler will complain at you if a new variant is added.